src/renameObjectKey.ts

```diff
@@ -0,0 +1,23 @@
+// renames a key while trying to preserve key ordering
+export const renameObjectKey = (obj: object, oldKey: string, newKey: string) => {
+  if (!obj || !obj[oldKey] || oldKey === newKey) {
+    return obj;
+  }
+
+  const values = [];
+  for (const key in obj) {
+    if (key !== newKey) {
+      values.push({
+        name: key === oldKey ? newKey : key,
+        value: obj[key],
+      });
+    }
+  }
+
+  const newObj = {};
+  for (const value of values) {
+    newObj[value.name] = value.value;
+  }
+
```

Comment:
Even shorter:

```
if (!obj || !obj[oldKey] || oldKey === newKey) {
  return obj;
}

const newObj = {};
for (const [key, value] of Object.entries(obj)) {
  if (key === oldKey) {
    newObj[newKey] = value;
  } else {
    newObj[key] = value;
  }
}

return newObj;
```

Reply:
Had to change it a bit to make one test case pass.